x/participation/keeper/msg_participate.go

```diff
@@ -13,16 +13,41 @@ import (
 func (k msgServer) Participate(goCtx context.Context, msg *types.MsgParticipate) (*types.MsgParticipateResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	// TODO: Read available allocation
+	availableAlloc, err := k.GetAvailableAllocations(ctx, msg.Participant)
+	if err != nil {
+		return nil, err
+	}
 
+	// check if auction exists
 	_, found := k.fundraisingKeeper.GetAuction(ctx, msg.AuctionID)
 	if !found {
 		return nil, sdkerrors.Wrapf(types.ErrAuctionNotFound, "auction %d not found", msg.AuctionID)
 	}
 
+	// check if a bid already exists
+	_, found = k.GetAuctionUsedAllocations(ctx, msg.Participant, msg.AuctionID)
+	if found {
+		return nil, sdkerrors.Wrapf(types.ErrInvalidBidder,
```

Comment:
`types.ErrInvalidBidder` we should not consider the notion of `Bidder` at this point

Also the error would be in my opinion more `ErrAlreadyParticipating`